File: Data Structures & Algorithms/simplify-path/submission-0.cs
public class Solution {
    public string SimplifyPath(string path) {
        Stack<string> simplified = new Stack<string>();
        List<string> splitPath = path.Split('/').ToList();
        foreach (string s in splitPath) {
            if (string.IsNullOrEmpty(s) || s.Equals(".")) {
                continue;
            }
            else if (s.Equals("..")) {
                if (simplified.Count > 0) {
                    simplified.Pop();
                }
            }
            else {
                simplified.Push(s);
            }
        }

        List<string> result = new List<string>();
        while (simplified.Count > 0) {
            result.Add(simplified.Pop());
        }

        result.Reverse();
        return "/" + string.Join("/", result);
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.java
class Solution {
    public int evalRPN(String[] tokens) {

        Stack<Integer> operation = new Stack<>();
        for (String s : tokens) {
            if (s.equals("+")) {
                operation.push(operation.pop() + operation.pop());
            }
            else if (s.equals("-")) {
                int a = operation.pop();
                int b = operation.pop();
                operation.push(b - a);
            }
            else if (s.equals("*")) {
                operation.push(operation.pop() * operation.pop());
            }
            else if (s.equals("/")){
                int a = operation.pop();
                int b = operation.pop();
                operation.push(b / a);
            }
            else {
                operation.push(Integer.parseInt(s));
            }
        }
        return operation.pop();
    }
}
